src/main/java/com/hillel/items_exchange/service/impl/CategoryServiceImpl.java

```diff
@@ -1,31 +1,242 @@
 package com.hillel.items_exchange.service.impl;
 
 import com.hillel.items_exchange.dao.CategoryRepository;
+import com.hillel.items_exchange.dao.SubcategoryRepository;
+import com.hillel.items_exchange.dto.CategoryControllerDto;
+import com.hillel.items_exchange.dto.SubcategoryControllerDto;
+import com.hillel.items_exchange.mapper.CategoryMapper;
+import com.hillel.items_exchange.model.Category;
+import com.hillel.items_exchange.model.Subcategory;
 import com.hillel.items_exchange.service.CategoryService;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.modelmapper.TypeToken;
 import org.springframework.stereotype.Service;
 
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
 public class CategoryServiceImpl implements CategoryService {
 
     private final CategoryRepository categoryRepository;
+    private final SubcategoryRepository subcategoryRepository;
+    private final ModelMapper modelMapper;
+    private final CategoryMapper categoryMapper;
 
     @Override
     public List<String> findAllCategoryNames() {
-        List<String> categoryNames = categoryRepository.findAllCategories();
+        List<String> categoryNames = categoryRepository.findAllCategoriesNames();
 
         if (categoryNames.isEmpty()) {
-            log.warn("IN CategoryServiceImpl (findAllCategoryNames): there are no categories");
+            log.warn("IN CategoryServiceImpl (findAllCategoryNames): there are not categories names");
             return Collections.singletonList("No categories");
         } else {
             log.info("IN CategoryServiceImpl (findAllCategoryNames): there are these categories: {}", categoryNames);
             return categoryNames;
         }
     }
+
+    @Override
+    public List<CategoryControllerDto> findAllCategories() {
+        List<CategoryControllerDto> allCategories = modelMapper.map(categoryRepository.findAll(),
+                new TypeToken<List<CategoryControllerDto>>() {
+                }.getType());
+
+        if (allCategories.isEmpty()) {
+            log.warn("IN CategoryServiceImpl (findAllCategories): there are not categories");
+            return Collections.emptyList();
+        } else {
+            log.info("IN CategoryServiceImpl (findAllCategories): there are these categories have been found: {}",
+                    allCategories);
+            return modelMapper.map(categoryRepository.findAll(), new TypeToken<List<CategoryControllerDto>>() {
+            }.getType());
+        }
+    }
+
+    @Override
+    public Optional<CategoryControllerDto> findCategoryById(Long id) {
+        if (categoryRepository.findById(id).isPresent()) {
```

Comment:
If you want to log do it in lambda, otherwise you don't use all features of optional.